src/machine.rs

```diff
@@ -84,9 +84,12 @@ pub struct MemoryExtra {
     /// An allocation ID to report when it is being allocated
     /// (helps for debugging memory leaks).
     tracked_alloc_id: Option<AllocId>,
+
+    /// Place where the `environ` static is stored. Its value should not change after initialization.
```

Comment:
I just realized this is potentially confusing... the place should not change, but the value stored in the place can change of course!
```suggestion
    /// Place where the `environ` static is stored. Lazily initialized, but then never changes.
```